src/pubsub/ua_pubsub_reader.c

```diff
@@ -25,9 +25,15 @@
 #include "ua_types_encoding_binary.h"
 #endif
 
+#ifdef UA_ENABLE_PUBSUB_BUFMALLOC
+#include "ua_pubsub_bufmalloc.h"
+#endif
+
 #define UA_MAX_SIZENAME           64  /* Max size of Qualified Name of Subscribed Variable */
 #define MIN_PAYLOAD_SIZE_ETHERNET 46
 
+#define READERGROUP_BUFFER_SIZE   4096
```

Comment:
If the buffer has a fixed size anyway, then I prefer a thread-local static buffer instead of the heap.
Less memory overhead per connection.
And less pressure on the cpu cache.